### PR TITLE
fix(e2e): compatibility e2e test fixes

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/compatibility-test-matrices/main/ica-unordered-channel-chain-a.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (crodriguezvega) [inline .github/compatibility-test-matrices/main/ica-unordered-channel-chain-a.json:1]: I added new files for compatibility testing of unordered ICA channels.

**Comment by crodriguezvega** [comment]
[Compatibility tests green](https://github.com/cosmos/ibc-go/actions/runs/7693055197) for `release/v8.1.x`. I will run them as well for the other release branches.

**Comment by damiannolan** [comment]
I'm currently looking at moving the conditionals for unknown fields to `sanitize.Messages`

ref: https://github.com/cosmos/ibc-go/blob/main/e2e/testsuite/sanitize/messages.go#L34

**Comment by damiannolan** [comment]
Changes included in https://github.com/cosmos/ibc-go/pull/5761/commits/c949d5b02b5576ba6d2064879ee9f02b33139776. I think it significantly reduced the diff. I ran locally using the regular `ORDERED` test and `UNORDERED` test against `main` tags.

**Review by damiannolan — APPROVED** [review]
Let me know if the changes look good to you @crodriguezvega! I'll drop my approval on this PR, happy to get it merged.

**Comment by crodriguezvega** [comment]
Thank you so much for the improvements, @damiannolan! Changes look great, I triggered a compatibility test run [here](https://github.com/cosmos/ibc-go/actions/runs/7696366544). I am happy to merge if they all get 🟢.

**Comment by damiannolan** [comment]
> Thank you so much for the improvements, @damiannolan! Changes look great, I triggered a compatibility test run [here](https://github.com/cosmos/ibc-go/actions/runs/7696366544). I am happy to merge if they all get 🟢.

I see some failures in gov/groups tests, due to the nested messaging. I think we can sanitize before putting into the proposals also. I can look into pushing a fix later if nobody gets around it.

**Comment by crodriguezvega** [comment]
[ICA test with gov and groups failed](https://github.com/cosmos/ibc-go/actions/runs/7696366544), but I added [sanitation of the messages contained in the proposal](https://github.com/cosmos/ibc-go/pull/5761/commits/2538051c963e0da27b9ca2dc749e2d791273e0d9) now, so I will trigger another [run](https://github.com/cosmos/ibc-go/actions/runs/7697211985).

**Comment by crodriguezvega** [comment]
[Test run was green](https://github.com/cosmos/ibc-go/actions/runs/7697211985), will merge as soon as I have a chance with updating the branch all the time!!

**Pull request merged**